src/passes/OptimizeInstructions.cpp

```diff
@@ -2891,6 +2891,17 @@ struct OptimizeInstructions
             }
           }
         }
+        if (unary->op == EqZInt32 || unary->op == EqZInt64) {
+          if (auto* c = getFallthrough(unary->value)->dynCast<Const>()) {
+            if (c->value.isZero()) {
+              return getDroppedChildrenAndAppend(unary,
+                                                 Literal::makeOne(Type::i32));
+            } else {
+              return getDroppedChildrenAndAppend(unary,
+                                                 Literal::makeZero(Type::i32));
+            }
```

Comment:
I think this can be simplified:
```suggestion
            return getDroppedChildrenAndAppend(unary,
                                               Literal::makeFromInt32(c->value.isZero(), Type::i32));
```